### PR TITLE
Closes #72, fixed supertest's request.head() webhook check invocation

### LINKED ISSUE #72: Error message "Webhook seems invalid or down" despite accessible URL
Hi there!

I'm using mailin version 3.0.1 from command line, like this:

```
mailin --webhook http://localhost:3000/emails
```

When it starts up, a warning is shown:
```
warn: Webhook http://localhost:3000/emails seems invalid or down. You may want to double check the webhook url.
```

I've curled that URL from the machine mailin is running, and it seems ok.
```
curl -X HEAD -I http://localhost:3000/emails

HTTP/1.1 200 OK
X-Powered-By: Express
Date: Tue, 15 Dec 2015 11:08:39 GMT
Connection: keep-alive
```
So, I started mailin with node-debug and set a breakpoint at [this line of code](https://github.com/Flolagale/mailin/blob/9ffad4388fde31fd595d9579b8b2125704f5fcb9/lib/mailin.js#L112):
```
node-debug /usr/lib/node_modules/mailin/cli.js --webhook http://localhost:3000/emails --port 2500
```

I've found this:
```
/* Check the webhook validity. */
    if (!configuration.disableWebhook) {
        request.head({
            url: configuration.webhook,
            timeout: 3000
        }, function (err, resp) {
            if (err || resp.statusCode !== 200) {
                logger.warn('Webhook ' + configuration.webhook +
                    ' seems invalid or down. You may want to double check the webhook url.');
            } else {
                logger.info('Webhook ' + configuration.webhook + ' is valid, up and running.');
            }
        });
    }
```

In spite of `configuration.webhook` being Ok, `function` is called with this `err`:

image

It looks like `superagent` is calling `127.0.0.1:80` instead of what's in `configuration.webhook`. S

I've taken a look at [current superagent docs](http://visionmedia.github.io/superagent/#request-basics), and it seems requests are done somewhat differently.

```javascript
request
  .head('/favicon.ico')
  .end(function(err, res){

  });
```
Maybe there's been a change in `superagent` API.

Shall I fork+pull request?

### PR DESCRIPTION
**Pull request by mefernandez**
Just 2 things in this PR:
1. A simple fix to supertest's `req.head()` call
2. Some changes in `test/mailinSpec.js` to check it's working.

Item 1 is a no-brainer, but item 2 should arise some discussion. From my point of view, the specs are mixing unit and integration tests, which makes testing very difficult. I'm not happy with the way I changed `mailinSpec.js` at all, but I couldn't find a more simple and elegant solution. Do you see any other way to do it?

### CODE REVIEW

File: test/mailinSpec.js [hunk 1]
```diff
@@ -16,15 +16,34 @@ chai.use(chaiAsPromised);
 var should = null;
 should = chai.Should();
 
+var server = express(),
```
Comment (mefernandez) [inline test/mailinSpec.js]: I had to move this up here, to make it listen before `Mailin.start()` gets called and checks `/webhook`.

File: test/mailinSpec.js [hunk 2]
```diff
@@ -16,15 +16,34 @@ chai.use(chaiAsPromised);
...
-    mailin.start({
-        // verbose: true,
-        smtpOptions: {
-            secure: false
-        }
-    }, function (err) {
+    /* Make an http server to receive the webhook. */
+    server.head('/webhook', function (req, res) {
+        res.send(200);
+        doing--;
```
Comment (mefernandez) [inline test/mailinSpec.js]: This one actually set `doing` to `-1`, for it is called before the test itself can set it doing to the required value.

File: test/mailinSpec.js [hunk 3]
```diff
@@ -36,7 +55,7 @@ describe('Mailin', function () {
     it('should post a json to a webhook after receiving an email and trigger some events', function (done) {
         this.timeout(30000);
 
-        var doing = 3; // Number of async operations we need to wait for before calling done
+        doing += 4; // Number of async operations we need to wait for before calling done
```
Comment (mefernandez) [inline test/mailinSpec.js]: Since `doing` is set in `before()`, I have to add to it instead of setting a fixed value.

**Comment by Flolagale** [comment]
@mefernandez awesome. Could you please merge master into this one to allow Travis to run? I just fixed the `.travis.yml`file.

**Comment by mefernandez** [comment]
@Flolagale Done. Seems Travis is passing tests Ok.

**Pull request merged**